pkg/runtime/k0s/k0sctl/v1beta1/types.go

```diff
@@ -0,0 +1,64 @@
+// Copyright © 2022 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	"github.com/k0sproject/k0sctl/pkg/apis/k0sctl.k0sproject.io/v1beta1/cluster"
+)
+
+// APIVersion is the current api version
+const APIVersion = "k0sctl.k0sproject.io/v1beta1"
+
+// ClusterMetadata defines cluster metadata
+type ClusterMetadata struct {
+	Name       string `yaml:"name" validate:"required" default:"k0s-cluster"`
+	Kubeconfig string `yaml:"-"`
+}
+
+// Cluster describes launchpad.yaml configuration
+type Cluster struct {
```

Comment:
I am afraid that we do not think this struct definition. Just make 
```
type K0sConfig struct{
         APIVersion string           `yaml:"apiVersion"`
	Kind       string           `yaml:"kind"`
	Metadata   *ClusterMetadata `yaml:"metadata"`
	Spec       *cluster.Spec    `yaml:"spec"`
}
```
?

Reply:
This struct is copy from k0s project. if we need to change it?